analysis/source/prepare_data/SConscript

```diff
@@ -1,7 +1,22 @@
 Import('env')
 
-env.BuildStata(target  = ['#%s/data.txt'       % env['PATHS']['build']['prepare_data']],
-               source  = ['#%s/create_data.do' % env['PATHS']['source']['prepare_data']],
-               log_ext =  'create_data')
+build  = env['PATHS']['build']['prepare_data']
+source = env['PATHS']['source']['prepare_data']
 
-env.Install('#release/lg', '#%s/data.txt' % env['PATHS']['build']['prepare_data'])
+env.BuildPython(target  = ['#%s/data.txt'       % build],
+                source  = ['#%s/create_data.py' % source],
+                log_ext =  'create_data')
+
+# env.BuildStata(target  = ['#%s/data_stata.txt' % build],
+#                source  = ['#%s/create_data.do' % source],
+#                log_ext =  'create_data_stata')
+
+# env.BuildR(target  = ['#%s/data_r.txt'    % build],
+#            source  = ['#%s/create_data.r' % source],
+#            log_ext =  'create_data_r')
+
+# env.BuildMatlab(target  = ['#%s/data_matlab.txt' % build],
+#                 source  = ['#%s/create_data.m'   % source],
+#                 log_ext =  'create_data_matlab')
+
+env.Install('#release/lg', '#%s/data.txt' % build)
```

Comment:
@yuchuan2016 two things:

1) I think it was MG's intention to only have this kind of demonstration in one subdirectory. I vote it be `prepare_data` over `analysis`.

2) If that's the case, do you see it being costly to have the same `data.txt` output for all three sources instead of `data_SUFFIX.txt`? I think that'll better demonstrate our point that we have different builders to accomplish the same thing.

Reply:
I have tried. The problem is that if you turn on two softwares, say, uncomment both `BuildPython` and `BuildStata`, these two commands would have the same target, and `scons` would fail if two command have the same target.

Reply:
I understand, but then you are forced to have both `Matlab` scripts turned on in descriptive and prepare_data. I don't think it's a big deal if we want to keep different softwares in both prepare_data and descriptive, but if we only keep it in prepare_data then we should have the same target and then only allow one to be turned on at each time.

Reply:
I see! I have changed the name and make a note in README.